cotk/models/ngram_language_model.py

```diff
@@ -205,6 +205,14 @@ def sent_log_prob(self, sent):
 			log_prob += np.log(self.score(sent_now[i + self.order], tuple(sent_now[i + 1:i + self.order])))
 		return log_prob
 
+	@classmethod
+	def _set_language_model(cls, language_model):
```

Comment:
Don't initialize with self, that means the program has to clone the whole object and its property, including self.dm and the whole dataset. Just initialize what you will use, e.g., self.left_pad_symbol. It will significantly improve performance.